### ISSUE TITLE
Heartbeat alarms in the components

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#8263]
@ticoann @scarletnorberg

please update this issue everytime you see a heartbeat alarm and try to provide the following info:
agent name, component name, heartbeat message, component status (running or not?), process status (PID existent or not). I guess I didn't miss anything...

Let me get started with vocms0250 (and 251 and 253), PhEDExInjector, hearbeat message was around 70min and the component was up & running. Problem with cmsweb phedex that was timing out.

**Comment by scarletnorberg** [comment]
Monday their were two agents that went down. One agent I did not catch vocms0250.cern.ch WorkQueueManagerWorkQueueManager and vocms0251 WorkQueueManagerWorkQueueManager. I only saw 251.
I checked twice and the log was still filling here:
2017-10-24 00:22:21,233:140205844248320:INFO:WMBSHelper:WMBS ACDC Parent File: MCFakeFile-vlimant_task_HIG-RunIISummer15wmLHEGS-01263__v1_T_170906_153317_9442-HIG-RunIISummer15wmLHEGS-01263_0-47c4fde91170d31450e7e64e8e96b2b2
2017-10-24 00:22:21,422:140205844248320:INFO:WMBSHelper:WMBS ACDC File: MCFakeFile-vlimant_task_HIG-RunIISummer15wmLHEGS-01263__v1_T_170906_153317_9442-HIG-RunIISummer15wmLHEGS-01263_0-47c4fde91170d31450e7e64e8e96b2b2
 on Location: set([u'T3_US_FNALLPC', u'T1_US_FNAL_Disk'])
2017-10-24 00:22:21,611:140205844248320:INFO:WMBSHelper:WMBS ACDC File: /store/unmerged/RunIISummer15wmLHEGS/ChargedHiggs_HplusTB_HplusToTauNu_IntermediateMassWithNeutral_M-145_13TeV_amcatnlo_pythia8/GEN-SIM/MCRUN2_71_V1-v3/50009/D6C366A7-F49C-E711-8A4E-0CC47A7C3428.root
 on Location: set([u'T3_US_FNALLPC', u'T1_US_FNAL_Disk'])
2017-10-24 00:22:21,616:140205844248320:INFO:WMBSHelper:WMBS ACDC Parent File: MCFakeFile-vlimant_task_HIG-RunIISummer15wmLHEGS-01263__v1_T_170906_153317_9442-HIG-RunIISummer15wmLHEGS-01263_0-47c4fde91170d31450e7e64e8e96b2b2
2017-10-24 00:22:22,280:140205844248320:INFO:WMBSHelper:WMBS ACDC File: MCFakeFile-vlimant_task_HIG-RunIISummer15wmLHEGS-01263__v1_T_170906_153317_9442-HIG-RunIISummer15wmLHEGS-01263_0-47c4fde91170d31450e7e64e8e96b2b2
 on Location: set([u'T3_US_FNALLPC', u'T1_US_FNAL_Disk'])
2017-10-24 00:22:22,360:140205852641024:INFO:WorkQueue:Acquired work for request(s): "vlimant_ACDC0_task_EXO-RunIISummer15wmLHEGS-04958__v1_T_171024_001725_1241".
We had at least one more on Tuesday but I saw it but had to go and never got to the log. I will try to be more diligent today.

**Comment by scarletnorberg** [comment]
vocms0250.cern.ch 
worker thread: WorkQueueManagerWMBSFileFeeder
status: Timeout
last updated: 2017/10/31 (Tue) 14:53:42 UTC
pid: 777368
error message:
Last worker thread heartbeat was 67 min ago
Has been updating the log:
017-10-31 17:07:49,883:140192458229504:INFO:WorkQueueManagerCleaner:Start updating & cleaning...
2017-10-31 17:08:31,275:140192483407616:INFO:WorkQueueManagerWorkPoller:Pulling work from https://cmsweb.cern.ch/couchdb/workqueue
2017-10-31 17:08:31,963:140192483407616:WARNING:WorkQueueManagerWorkPoller:Workqueue didn't pass the retrieve condition: NOT pulling work
2017-10-31 17:08:32,584:140192483407616:INFO:WorkQueueManagerWorkPoller:Obtained 0 unit(s) of work
2017-10-31 17:08:32,585:140192483407616:INFO:WorkQueueManagerWorkPoller:Splitting new work
2017-10-31 17:08:32,585:140192483407616:INFO:WorkQueue:fixing conflict...
2017-10-31 17:08:32,606:140192483407616:INFO:WorkQueueManagerWorkPoller:Splitting finished

**Comment by scarletnorberg** [comment]
vocms0250.cern.ch
JobAccountant
worker thread: JobAccountantPoller
status: Timeout
last updated: 2017/10/31 (Tue) 15:00:21 UTC
pid: 823957
error message:
Last worker thread heartbeat was 66 min ago
The log has been updated:
2017-10-31 17:10:35,690:139638918838016:INFO:AccountantWorker:Job 1492234 , handle successful job
2017-10-31 17:10:35,704:139638918838016:INFO:AccountantWorker:Handling /data/srv/wmagent/v1.1.6.patch4/install/wmagent/JobCreator/JobCache/pdmvserv_task_TOP-RunIISummer15wmLHEGS-00265__v1_T_171020_160806_3532/TOP-RunIISummer15wmLHEGS-00265_0/JobCollection_160809_0/job_1492235/Report.0.pkl
2017-10-31 17:10:35,727:139638918838016:INFO:AccountantWorker:Job 1492235 , handle successful job
2017-10-31 17:10:35,738:139638918838016:INFO:AccountantWorker:Handling /data/srv/wmagent/v1.1.6.patch4/install/wmagent/JobCreator/JobCache/pdmvserv_task_TOP-RunIISummer15wmLHEGS-00265__v1_T_171020_160806_3532/TOP-RunIISummer15wmLHEGS-00265_0/JobCollection_160809_0/job_1492236/Report.0.pkl

**Comment by amaltaro** [comment]
JobArchiver has a heartbeat timeout on vocms0253, "Last worker thread heartbeat was 41 min ago".
Log looks like
```
2017-11-01 13:07:11,495:140163607344896:INFO:JobArchiverPoller:Found 7405 finished jobs to archive
2017-11-01 13:47:53,536:140163607344896:INFO:JobArchiverPoller:Successfully archived 7405 jobs out of 7405.
2017-11-01 13:47:54,637:140163607344896:INFO:JobArchiverPoller:Found 4 filesets to be closed
2017-11-01 13:48:02,460:140163607344896:INFO:JobArchiverPoller:Found 0 workflows to mark as injected
2017-11-01 13:52:23,013:140163607344896:INFO:JobArchiverPoller:Found 10041 finished jobs to archive
2017-11-01 14:43:18,732:140163607344896:INFO:JobArchiverPoller:Successfully archived 10000 jobs out of 10041.
2017-11-01 14:43:34,907:140163607344896:INFO:JobArchiverPoller:Successfully archived 10041 jobs out of 10041.
2017-11-01 14:43:36,194:140163607344896:INFO:JobArchiverPoller:Found 6 filesets to be closed
2017-11-01 14:43:43,948:140163607344896:INFO:JobArchiverPoller:Found 0 workflows to mark as injected
2017-11-01 14:48:07,599:140163607344896:INFO:JobArchiverPoller:Found 9712 finished jobs to archive
2017-11-01 15:48:36,278:140163607344896:INFO:JobArchiverPoller:Successfully archived 9712 jobs out of 9712.
2017-11-01 15:48:37,632:140163607344896:INFO:JobArchiverPoller:Found 15 filesets to be closed
2017-11-01 15:48:45,797:140163607344896:INFO:JobArchiverPoller:Found 0 workflows to mark as injected
2017-11-01 15:53:12,263:140163607344896:INFO:JobArchiverPoller:Found 11810 finished jobs to archive
...
now is 16:38
```

so in average 2k jobs in 10min. We should either slice it or process not more than X jobs per cycle (component polling cycle is set to 4min).

**Comment by scarletnorberg** [comment]
agent: vocms0250.cern.ch (1.1.6.patch4)
agent last updated: 2017/11/9 (Thu) 04:19:05 UTC : 0 h 1 m
data last updated: N/A
status: Draining Agent; Components or Thread down;
team: production
component errors for: JobSubmitter,JobStatusLite
JobSubmitter
worker thread: JobSubmitterPoller
status: Timeout
last updated: 2017/11/8 (Wed) 15:16:27 UTC
pid: 3622717
error message:
Last worker thread heartbeat was 782 min ago
JobStatusLite
worker thread: StatusPoller
status: Timeout
last updated: 2017/11/9 (Thu) 03:39:12 UTC
pid: 777448
error message:
Last worker thread heartbeat was 39 min ago

Two components have heart beat issues:

JobStatusLie:
2017-11-09 04:37:53,831:140192464979712:ERROR:SimpleCondorPlugin:No job report for job with id 2230382 and gridid 16636.166
2017-11-09 04:37:53,835:140192464979712:ERROR:SimpleCondorPlugin:No job report for job with id 2230393 and gridid 16636.177
2017-11-09 04:37:53,840:140192464979712:ERROR:SimpleCondorPlugin:No job report for job with id 2230379 and gridid 16636.163
2017-11-09 04:37:53,844:140192464979712:ERROR:SimpleCondorPlugin:No job report for job with id 2230396 and gridid 16636.180
2017-11-09 04:39:12,573:140192464979712:INFO:StatusPoller:Running job status poller algorithm...
2017-11-09 04:39:17,008:140192464979712:INFO:BossAirAPI:About to start building running jobs
JobSubmit:
2017-11-08 16:20:10,918:139749410522880:ERROR:SimpleCondorPlugin:Failed to create new proc id.
Traceback (most recent call last):
  File "/data/srv/wmagent/v1.1.6.patch4/sw/slc7_amd64_gcc630/cms/wmagent/1.1.6.patch4/lib/python2.7/site-packages/WMCore/BossAir/Plugins/SimpleCondorPlugin.py", line 148, in submit
    clusterId = schedd.submitMany(clusterAd, procAds)
RuntimeError: Failed to create new proc id.
2017-11-08 16:20:10,919:139749410522880:INFO:SimpleCondorPlugin:Done submitting jobs for this cycle in SimpleCondorPlugin
2017-11-08 16:20:11,052:139749410522880:INFO:JobSubmitterPoller:Jobs that succeeded/failed submission: 0/924.
The logs look fine.

**Comment by scarletnorberg** [comment]
agent: vocms0251.cern.ch (1.1.6.patch4)
agent last updated: 2017/11/9 (Thu) 04:44:20 UTC : 0 h 0 m
data last updated: N/A
status: Draining Agent; Components or Thread down;
team: production
component errors for: JobStatusLite
JobStatusLite
worker thread: StatusPoller
status: Timeout
last updated: 2017/11/9 (Thu) 04:12:13 UTC
pid: 56322
error message:
Last worker thread heartbeat was 31 min ago
2017-11-09 05:10:57,471:140609077765888:ERROR:SimpleCondorPlugin:No job report for job with id 2491247 and gridid 22323.74
2017-11-09 05:11:12,448:140609077765888:INFO:StatusPoller:Killing job 1716791 because it has exceeded timeout for status 'Pending'
2017-11-09 05:11:12,511:140609077765888:INFO:StatusPoller:Killing job 1716691 because it has exceeded timeout for status 'Pending'
2017-11-09 05:11:12,630:140609077765888:INFO:SimpleCondorPlugin:Killing 2 jobs from the queue
2017-11-09 05:12:13,177:140609077765888:INFO:StatusPoller:Running job status poller algorithm...
2017-11-09 05:12:18,002:140609077765888:INFO:BossAirAPI:About to start building running jobs
Not sure about this one. It went green and is now red again.

**Comment by scarletnorberg** [comment]
agent: vocms0253.cern.ch (1.1.6.patch5)
agent last updated: 2017/11/16 (Thu) 05:46:28 UTC : 0 h 2 m
data last updated: N/A
status: Components or Thread down;
team: production
component errors for: WorkQueueManager
WorkQueueManager
worker thread: WorkQueueManagerWMBSFileFeeder
status: Timeout
last updated: 2017/11/15 (Wed) 23:53:24 UTC
pid: 878137
error message:
Last worker thread heartbeat was 352 min ago
2017-11-16 06:45:26,812:140436073084672:INFO:WorkQueueBackend:Getting up to 100 available work from https://cmsweb.cern.ch/couchdb/workqueue
2017-11-16 06:45:26,812:140436073084672:INFO:WorkQueueBackend:setting teams ['production']
2017-11-16 06:46:25,429:140436056299264:WARNING:SiteDB:No PSNs for PNN: T3_US_HEPCloud
2017-11-16 06:46:30,395:140436047906560:INFO:WorkQueueManagerCleaner:Start updating & cleaning...
2017-11-16 06:46:43,080:140436073084672:WARNING:WorkQueue:No available work in parent queue.
2017-11-16 06:46:43,843:140436073084672:INFO:WorkQueueManagerWorkPoller:Obtained 0 unit(s) of work
2017-11-16 06:46:43,844:140436073084672:INFO:WorkQueueManagerWorkPoller:Splitting new work
2017-11-16 06:46:43,844:140436073084672:INFO:WorkQueue:fixing conflict...
2017-11-16 06:46:43,864:140436073084672:INFO:WorkQueueManagerWorkPoller:Splitting finished
Seems to be fine.

**Comment by amaltaro** [comment]
Sigh, something funky is going on. I restarted the component and it has another hearbeat timeout for that thread. Investigating...

**Comment by amaltaro** [comment]
I've applied this temporary patch:
https://github.com/amaltaro/WMCore/commit/f001e8cf38b684bc67324925789c564db63ad2f9

to debug this issue.

It might be getting stuck in the wmbs operations for this request
```
2017-11-16 14:46:06,305:139850554177280:INFO:WorkQueue:AMR element for request: mcremone_ACDC0_task_HIG-RunIISummer15wmLHEGS-01737__v1_T_171115_163727_9771
```

though I don't see any problems with it, besides the MCFakeFile as parent, as Seangchan already noticed and created an issue.

**Comment by scarletnorberg** [comment]
ent: vocms0253.cern.ch (1.1.6.patch5)
agent last updated: 2017/11/17 (Fri) 15:09:13 UTC : 0 h 0 m
data last updated: N/A
status: Components or Thread down;
team: production
component errors for: WorkQueueManager
WorkQueueManager
worker thread: WorkQueueManagerWMBSFileFeeder
status: Timeout
last updated: 2017/11/17 (Fri) 09:48:48 UTC
pid: 2021375
error message:
Last worker thread heartbeat was 320 min ago

Log file:
2017-11-17 16:09:58,624:140526496847616:INFO:WorkQueueBackend:Getting up to 100 available work from https://cmsweb.cern.ch/couchdb/workqueue
2017-11-17 16:09:58,624:140526496847616:INFO:WorkQueueBackend:setting teams ['production']
2017-11-17 16:10:42,215:140526471669504:INFO:WorkQueueManagerCleaner:Start updating & cleaning...
2017-11-17 16:11:22,691:140526480062208:INFO:WorkQueueManagerLocationPoller:Updating data locations
2017-11-17 16:11:32,874:140526496847616:WARNING:WorkQueue:No available work in parent queue.
2017-11-17 16:11:33,093:140526496847616:INFO:WorkQueueManagerWorkPoller:Obtained 0 unit(s) of work
2017-11-17 16:11:33,093:140526496847616:INFO:WorkQueueManagerWorkPoller:Splitting new work
2017-11-17 16:11:33,094:140526496847616:INFO:WorkQueue:fixing conflict...
2017-11-17 16:11:33,110:140526496847616:INFO:WorkQueueManagerWorkPoller:Splitting finished
Seems to be updating.

**Comment by scarletnorberg** [comment]
agent: vocms0251.cern.ch (1.1.6.patch4)
agent last updated: 2017/11/20 (Mon) 04:20:00 UTC : 0 h 0 m
data last updated: N/A
status: Draining Agent; Components or Thread down;
team: production
component errors for: AgentStatusWatcherAgentStatusWatcher
worker thread: DrainStatusPoller
status: Timeout
last updated: 2017/11/20 (Mon) 03:03:48 UTC
pid: 4051645
error message:Last worker thread heartbeat was 76 min ago
2017-11-20 05:19:57,556:140218899035904:INFO:AgentStatusPoller:Getting agent info ...
2017-11-20 05:19:59,167:140218899035904:INFO:AgentStatusPoller:Agent is in DrainMode
2017-11-20 05:19:59,226:140218899035904:INFO:AgentStatusPoller:CouchDB is running with 8 processes
2017-11-20 05:19:59,227:140218899035904:INFO:AgentStatusPoller:List of agent components down: ['AgentStatusWatcher']
2017-11-20 05:19:59,438:140218899035904:INFO:AgentStatusPoller:Getting wmbs job info ...
2017-11-20 05:20:00,958:140218899035904:INFO:AgentStatusPoller:WMBS data collected in: 1 secs
2017-11-20 05:20:00,975:140218899035904:INFO:AgentStatusPoller:Local WorkQueue data collected in: 0 secs

**Comment by scarletnorberg** [comment]
This is after the oracle issue
agent: vocms0254.cern.ch (1.1.6.patch6)
agent last updated: 2017/11/20 (Mon) 03:01:27 UTC : 1 h 31 m
data last updated: N/A
status: Data is not updated: AnalyticsDataCollector Down; Components or Thread down;
team: production
component errors for: JobAccountant
JobAccountant
worker thread: JobAccountantPoller
status: Timeout
last updated: 2017/11/20 (Mon) 02:28:04 UTC
pid: 2483059
error message:
Last worker thread heartbeat was 33 min ago

2017-11-20 05:32:47,282:140501923141376:INFO:AccountantWorker:Job 1233805 , handle successful job
2017-11-20 05:32:47,289:140501923141376:INFO:AccountantWorker:Handling /data/srv/wmagent/v1.1.6.patch6/install/wmagent/JobCreator/JobCache/pdmvserv_task_HIG-RunIISummer15wmLHEGS-02005__v1_T_171117_134448_1564/HIG-RunIISummer15wmLHEGS-02005_0/JobCollection_90438_0/job_1233830/Report.0.pkl
2017-11-20 05:32:47,298:140501923141376:INFO:AccountantWorker:Job 1233830 , handle successful job
2017-11-20 05:32:47,311:140501923141376:INFO:AddRunLumi:AMR received 16 file entries
2017-11-20 05:32:47,311:140501923141376:INFO:AddRunLumi:AMR inserting 80 lumis from sliceBinds
2017-11-20 05:32:47,415:140501923141376:INFO:DashboardReporter:Handling 8 jobs
2017-11-20 05:33:48,490:140501923141376:INFO:JobAccountantPoller:Found 0 completed jobs

**Comment by scarletnorberg** [comment]
This is after the oracle issue
agent: vocms0258.cern.ch (1.1.6.patch4)
agent last updated: 2017/11/20 (Mon) 03:02:00 UTC : 1 h 33 m
data last updated: N/A
status: Data is not updated: AnalyticsDataCollector Down;
team: relval

2017-11-20 05:32:14,393:140633064466176:INFO:AnalyticsPoller:Getting Job Couch Data ...
2017-11-20 05:32:14,707:140633064466176:INFO:DataCollectAPI:Found 183 requests
2017-11-20 05:32:14,708:140633064466176:INFO:AnalyticsPoller:Getting FWJRJob Couch Data ...
2017-11-20 05:32:14,911:140633064466176:INFO:AnalyticsPoller:Getting Batch Job Data ...
2017-11-20 05:32:14,995:140633064466176:INFO:AnalyticsPoller:Getting Finished Task Data ...
2017-11-20 05:32:15,189:140633064466176:INFO:AnalyticsPoller:Getting DBS PhEDEx upload status ...
2017-11-20 05:32:15,278:140633064466176:INFO:AnalyticsPoller:Getting Local Queue Data ...
2017-11-20 05:32:15,287:140633064466176:INFO:AnalyticsPoller:Combining data from
                                   Job Couch(183),
                                   FWJR(153),
                                   WorkflowsWithSkippedFile(10),
                                   Batch Job(63),
                                   Finished Tasks(204),
                                   Local Queue(68)
                                   Completed workflows(137)..  ...
2017-11-20 05:32:15,289:140633064466176:INFO:AnalyticsPoller:204 requests Data combined,
 uploading request data...
2017-11-20 05:32:16,504:140633064466176:INFO:AnalyticsPoller:Request data upload success
 204 request,
sleep for next cycle

**Comment by scarletnorberg** [comment]
This is after the oracle issue
agent: vocms0310.cern.ch (1.1.4.patch2)
agent last updated: 2017/11/20 (Mon) 03:01:46 UTC : 1 h 40 m
data last updated: N/A
status: Draining Agent; Data is not updated: AnalyticsDataCollector Down;
team: production

2017-11-20 05:35:49,551:139895037478656:INFO:AnalyticsPoller:Getting DBS PhEDEx upload status ...
2017-11-20 05:35:49,557:139895037478656:INFO:AnalyticsPoller:Getting Local Queue Data ...
2017-11-20 05:35:49,563:139895037478656:INFO:AnalyticsPoller:Combining data from
                                   Job Couch(4),
                                   FWJR(4),
                                   WorkflowsWithSkippedFile(0),
                                   Batch Job(1),
                                   Finished Tasks(5),
                                   Local Queue(1)
                                   Completed workflows(5)..  ...
2017-11-20 05:35:49,563:139895037478656:INFO:AnalyticsPoller:5 requests Data combined,
 uploading request data...
2017-11-20 05:35:49,664:139895037478656:INFO:AnalyticsPoller:Request data upload success
 5 request,
sleep for next cycle

**Comment by amaltaro** [comment]
Scarlet, where these not running anymore?
I picked DrainStatusPoller in vocms0251, and the thread had been terminated (oracle issue). I guess you didn't restart AgentStatusWatcher on that agent?

About WorkQueueManagerWMBSFileFeeder in vocms0253, we were working on a patch since the end of the last week, but we might push that in only for the new agent (the component is back to normal business now).

About vocms0254, the agent has AnalyticsDataCollector and AgentStatusWatcher down. That's why JobAccountant is said as to be down (though it's running without any issues). Restarting those. Actually, there were more components down, but since AgentStatusWatcher was not running, it wasn't showing up in wmstats

**Comment by scarletnorberg** [comment]
vocms0251 When I checked it I did not even try restarting it. The log file posted above is what I looked at it was running it stop why it stopped I am not sure but it was not me.
Again if it was list on this it said heart beat issue when I checked the log file the component was running.

**Comment by scarletnorberg** [comment]
agent: vocms0251.cern.ch (1.1.6.patch4)
agent last updated: 2017/11/22 (Wed) 16:31:12 UTC : 0 h 3 m
data last updated: N/A
status: Draining Agent; Components or Thread down;
team: production
component errors for: TaskArchiver
TaskArchiver
worker thread: CleanCouchPoller
status: Timeout
last updated: 2017/11/22 (Wed) 15:20:59 UTC
pid: 4051565
error message:
Last worker thread heartbeat was 70 min ago
drain statistics:
workflows completed (false)
The component is still updating:
2017-11-22 17:35:02,439:140218892261120:INFO:CleanCouchPoller:Processing 8 errors, for job id 931880
2017-11-22 17:35:02,763:140218892261120:INFO:CleanCouchPoller:Processing 8 errors, for job id 931882
2017-11-22 17:35:03,086:140218892261120:INFO:CleanCouchPoller:Processing 8 errors, for job id 931885
2017-11-22 17:35:03,421:140218892261120:INFO:CleanCouchPoller:Processing 8 errors, for job id 931886
2017-11-22 17:35:03,806:140218892261120:INFO:CleanCouchPoller:Processing 8 errors, for job id 931888
2017-11-22 17:35:04,142:140218892261120:INFO:CleanCouchPoller:Processing 8 errors, for job id 931889
2017-11-22 17:35:04,501:140218892261120:INFO:CleanCouchPoller:Processing 8 errors, for job id 931891

**Comment by scarletnorberg** [comment]
agent: vocms0253.cern.ch (1.1.6.patch5)
agent last updated: 2017/11/29 (Wed) 14:06:19 UTC : 0 h 4 m
data last updated: N/A
status: Draining Agent; Components or Thread down;
team: production
component errors for: JobAccountant
JobAccountant
worker thread: JobAccountantPoller
status: Timeout
last updated: 2017/11/27 (Mon) 19:57:30 UTC
pid: 815048
error message:
Last worker thread heartbeat was 2528 min ago
drain statistics:
workflows completed (false)

Last night:
2017-11-28 21:59:52,764:139969730012928:INFO:AccountantWorker:Handling /data/srv/wmagent/v1.1.6.patch5/install/wmagent/JobCreator/JobCache/vlimant_task_MUO-RunIIFall17GS-00013__v1_T_171030_094645_4523/MUO-RunIIFall17GS-00013_0/JobCollection_281997_0/job_3153149/Report.0.pkl
2017-11-28 21:59:52,791:139969730012928:INFO:AccountantWorker:Job 3153149 , handle successful job
2017-11-28 21:59:52,805:139969730012928:INFO:AccountantWorker:Handling /data/srv/wmagent/v1.1.6.patch5/install/wmagent/JobCreator/JobCache/vlimant_task_MUO-RunIIFall17GS-00013__v1_T_171030_094645_4523/MUO-RunIIFall17GS-00013_0/JobCollection_281997_0/job_3153150/Report.0.pkl
2017-11-28 21:59:52,846:139969730012928:WARNING:AccountantWorker:Job 3153150 , bad jobReport, failing job
2017-11-28 21:59:52,856:139969730012928:INFO:AccountantWorker:Handling /data/srv/wmagent/v1.1.6.patch5/install/wmagent/JobCreator/JobCache/vlimant_task_MUO-RunIIFall17GS-00013__v1_T_171030_094645_4523/MUO-RunIIFall17GS-00013_0/JobCollection_281997_0/job_3153151/Report.0.pkl
2017-11-28 21:59:52,881:139969730012928:INFO:AccountantWorker:Job 3153151 , handle successful job
2017-11-28 21:59:52,924:139969730012928:INFO:AddRunLumi:AMR received 97 file entries
2017-11-28 21:59:52,925:139969730012928:INFO:AddRunLumi:AMR inserting 97 lumis from sliceBinds
2017-11-28 21:59:53,975:139969730012928:INFO:DashboardReporter:Handling 97 jobs

This morning:
2017-11-29 15:04:48,351:139969730012928:INFO:AccountantWorker:Job 3313446 , handle successful job
2017-11-29 15:04:48,358:139969730012928:INFO:AccountantWorker:Handling /data/srv/wmagent/v1.1.6.patch5/install/wmagent/JobCreator/JobCache/vlimant_task_MUO-RunIIFall17GS-00013__v1_T_171030_094645_4523/MUO-RunIIFall17GS-00013_0/JobCollection_282353_0/job_3313447/Report.0.pkl
2017-11-29 15:04:48,367:139969730012928:INFO:AccountantWorker:Job 3313447 , handle successful job
2017-11-29 15:04:48,393:139969730012928:INFO:AddRunLumi:AMR received 100 file entries
2017-11-29 15:04:48,393:139969730012928:INFO:AddRunLumi:AMR inserting 100 lumis from sliceBinds
2017-11-29 15:04:49,248:139969730012928:INFO:DashboardReporter:Handling 100 jobs

**Comment by scarletnorberg** [comment]
Yes I have heard it was more it had not been red and now is red so just keeping tack of things.

**Comment by scarletnorberg** [comment]
In principle I am not sure if this is a maybe we should put a longer heart beat or if this is the case that is odd that may not happen again (but probable will happen again) and we just will have a red agent for a while, while it catches up.

**Comment by ticoann** [comment]
@amaltaro, Alan do you want to still keep this open?

**Comment by amaltaro** [comment]
Just to wrap up:
 - WorkQueueManagerWMBSFileFeeder had issues related to too many parents (should have been fixed in 1.1.8.patch2)
 - JobAccountant is the most expensive component. Average of 10k docs per hour. Specific issue on #4656
 - JobArchiver is not very slow, but sometimes it triggers a heartbeat between 30 and 60 min. We might review and try to improve it eventually.
 - JobStatusLite falls on the same case of JobArchiver. Deserves a code refactoring eventually (which should buy us a better job tracking as well).

Seangchan, feel free to open any issues from this summary and/or just close this issue.

**Comment by ticoann** [comment]
Thanks Alan, I created the new issue  for performance bottleneck in components
